pkg/cli/run_test.go

```diff
@@ -28,100 +36,284 @@ func TestRunArgs_Env(t *testing.T) {
 	assert.Equal(t, "1", opts.Env[1].Value)
 }
 
-func TestRunMemory(t *testing.T) {
+func TestRun(t *testing.T) {
 	type fields struct {
-		All   bool
-		Type  []string
-		Force bool
+		Quiet  bool
+		Output string
+		All    bool
+		Force  bool
 	}
 	type args struct {
-		cmd    *cobra.Command
-		args   []string
-		client *testdata.MockClient
+		cmd  *cobra.Command
+		args []string
 	}
-	var _, w, _ = os.Pipe()
 	tests := []struct {
-		name           string
-		fields         fields
-		args           args
-		wantErr        bool
-		wantOut        string
-		commandContext CommandContext
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+		wantOut string
+		prepare func(f *mocks.MockClient)
 	}{
 		{
 			name: "acorn run -h", fields: fields{
 				All:   false,
-				Type:  nil,
 				Force: true,
 			},
-			commandContext: CommandContext{
-				ClientFactory: &testdata.MockClientFactory{},
-				StdOut:        w,
-				StdErr:        w,
-				StdIn:         strings.NewReader(""),
-			},
+
 			args: args{
-				args:   []string{"-h"},
-				client: &testdata.MockClient{},
+				args: []string{"-h"},
 			},
 			wantErr: false,
 			wantOut: "./testdata/run/acorn_run_help.txt",
 		},
 		{
 			name: "acorn run -m found.container=256Miii found ", fields: fields{
 				All:   false,
-				Type:  nil,
 				Force: true,
 			},
-			commandContext: CommandContext{
-				ClientFactory: &testdata.MockClientFactory{},
-				StdOut:        w,
-				StdErr:        w,
-				StdIn:         strings.NewReader(""),
-			},
 			args: args{
-				args:   []string{"-m found.container=256Miii", "found"},
-				client: &testdata.MockClient{},
+				args: []string{"-m found.container=256Miii", "found"},
+			},
+			prepare: func(f *mocks.MockClient) {
+				f.EXPECT().AppGet(gomock.Any(), gomock.Any()).DoAndReturn(
+					func(ctx context.Context, name string) (*apiv1.App, error) {
```

Comment:
The logic of this function appears to be the same in all 5 test cases. What do you think of creating a function and reusing it to cut down on the code a bit?